worlds/sc2/client_gui.py

```diff
@@ -463,6 +473,39 @@ def get_location_type_title(self, location_type: LocationType) -> str:
         else:
             title += ""
         return title
+    
+    def is_scoutable(self, remaining_locations, mission_available: bool, layout_locked: bool, campaign_locked: bool) -> bool:
+        if self.ctx.mission_order_scouting == 5:
+            return True
+        elif self.ctx.mission_order_scouting == 4 and not campaign_locked:
+            return True
+        elif self.ctx.mission_order_scouting == 3 and not layout_locked:
+            return True
+        elif self.ctx.mission_order_scouting == 2 and mission_available:
+            return True
+        elif self.ctx.mission_order_scouting == 1 and len([loc for loc in remaining_locations if loc[0] in (LocationType.VICTORY, LocationType.VICTORY_CACHE)]) == 0:
+            # Assuming that when a mission is completed, all victory location are removed 
+            return True
+        else:
+            return False
+
+
+    def handle_scout_display(self, location_name: str) -> str:
+        # Only one information is provided for the victory locations of a mission
+        if " Cache (" in location_name:
+            location_name = location_name.split(" Cache")[0]
+        item_classification_key = self.ctx.mission_item_classification[location_name]
+        if item_classification_key == 0:
+            return " [color=00EEEE](Filler)[/color]"
+        if item_classification_key == 1:
+            return " [color=AF99EF](Progression)[/color]"
+        elif item_classification_key == 2:
+            return " [color=6D8BE8](Usefull)[/color]"
+        elif item_classification_key == 3:
+            return " [color=FA8072](Trap)[/color]"
```

Comment:
As you insert the values as: `mission_item_classification[location_name] = loc.item.classification.as_flag()`

Could you compare to for example: `ItemClassification.filler.as_flag()`? (`as_flag()` drops the skip_balancing information that's not needed there)
Also Trap item should have value of 4 (see `ItemClassification`)

Reply:
Corrected Trap classification (tested with a game that has trap)

Changed hard-coded integer to `ItemClassification.XXX`. Did not use `.as_flag()` here since `mission_item_classification` is created using `as_flag()` and, as such, `skip_balancing` and `progression_skip_balancing` are impossible cases.